courses/fundamentals_of_ada/adv_240_tasking.rst

```diff
@@ -931,6 +1058,189 @@ Example: Main
 .. include:: examples/task_select_multiple_or/src/main.adb
     :code: Ada
 
+------
+Quiz
+------
+
+.. code:: Ada
+
+    task T is
+       entry E1;
+       entry E2;
+    end T;
+    ...
+    task body Other_Task is
+    begin
+       select
+          T.E1;
+       or
+          T.E2;
+       end select;
+    end Other_Task;
+
+What is the result of compiling and running this code?
+
+A. :ada:`T.E1` is called
+B. Nothing
+C. :answer:`Compilation error`
+D. Runtime error
+
+.. container:: animate
+
+    A :ada:`select` entry call can only call one :ada:`entry` at a time.
+
+------
+Quiz
+------
+
+.. container:: columns
+
+ .. container:: column
+
+  .. code:: Ada
+
+      procedure Main is
+         task T is
+            entry A;
+         end T;
+
+         task body T is
+         begin
+            select
+               accept A;
+               Put_Line ("A");
+            else
+               delay 1.0;
+            end select;
+         end T;
+      begin
+         select
+            T.A;
+         else
+            delay 1.0;
+         end select;
+      end Main;
+
+ .. container:: column
+
+  What is the output of this code?
+
+  A. "AAAAA..."
+  B. :answer:`Nothing`
+  C. Compilation error
+  D. Runtime error
+
+  .. container:: animate
+
+      Common mistake: :ada:`Main` and :ada:`T` won't wait on each other and
+      will both execute their :ada:`delay` statement only.
+
+.
+
+------
+Quiz
+------
+
+.. code:: Ada
+
+    procedure Main is
+       task type T is
+          entry A;
+       end T;
+
+       task body T is
+       begin
+          select
+             accept A;
+          or
+             terminate;
+          end select;
+
+          Put_Line ("Terminated");
+       end T;
+
+       My_Task : T;
+    begin
+       null;
+    end Main;
+
+What is the output of this code?
+
+A. "Terminated"
+B. :answer:`Nothing`
+C. Compilation error
+D. Runtime error
+
+.. container:: animate
+
+    :ada:`T` is terminated at the end of :ada:`Main`
+
+------
+Quiz
+------
+
+.. code:: Ada
+
+    procedure Main is
+    begin
+       select
+          delay 2.0;
+       then abort
+          loop
+             delay 1.5;
+             Put_Line ("A");
+          end loop;
+       end select;
+
+       Put_Line ("B");
+    end Main;
+
+What is the output of this code?
+
+A. "A"
+B. "AAAA..."
+C. :answer:`"AB"`
```

Comment:
Replace "put_line" with "put" otherwise the characters will be on separate lines